### ISSUE TITLE
Add course map filters for hiding completed items and certification only items

### ISSUE THREAD

**Issue by juggernate** [FreeCodeCamp/FreeCodeCamp#6186]
The course map is great, but is getting long. It would be great to have filters to hide completed items and hide non-certification items as well.

This may be a separate issue, but it would be nice to make the map traversal / next item take the filtering options into account (so it could skip completed items, this would be helpful when new items are added to the map in between existing completed items)

**Comment by Greenheart** [comment]
I like this idea. Maybe expand and minimize different sections and click to open them?

**Comment by SaintPeter** [comment]
I agree that some sort of filter would be nice.

Can you maybe brainstorm on what this might look like?  A drop-down?  A type-in box?

**Comment by juggernate** [comment]
I was imagining these as a row of buttons above the first map item that would always be present and switch with the state. I think at least the completion filter might be commonly used. 
- Show/Hide Completed
- Show/Hide Certification Only (or Show/Hide Non-Certification)
  - This could be combined with the "Required Courses disclaimer"
- Expand/Collapse All
  - While this is a different concept, it could be grouped with, or right in front of the others?

**Comment by SaintPeter** [comment]
We should have Expand/Collapse all coming in the next day or so.  It's done, just waiting on QA.
Show/Hide completed should be easy.
"Hide non-required", maybe?  That'd probably be doable. We have the info there, just need to build a class for it. 

These are all solid ideas.  I'll add it to the roadmap.

**Comment by juggernate** [comment]
Nice! Yeah, "Hide non-required" sounds clearer.
